samuroi/util/mask_generator/ilastik_functions.py

```diff
@@ -0,0 +1,80 @@
+__author__ = 'stephenlenzi'
+
+import os
+import h5py
+import numpy as np
+import tempfile
+import subprocess
+
+
+"""These are for running ilastik in headless mode within Python"""
+
+
+def get_ilastik_project_path():
+    for f in os.listdir("./"):
+        if f[-4:] == '.ilp':
+            return f
+        raise IOError('ilastik project file not found')
+
+
+def ilastik_segment(data, ilastik_path, ilastik_project_path=None):
+    """
+    Convert image data into a segmentation using ilastik. It requires that you have created a project file
+    specifically for the type of images you want to classify. If this function fails, you should first verify that
+    ilastik is able to form a segmentation when you add the image to the project file as you may need to train
+    the algorithm more.
+
+    :param data: NxM array, NxM is image shape.
+    :param ilastik_path: path to ilastik installation (should be named something like run_ilastik.sh)
+    :param ilastik_project_path:  path to ilastik project file (user must create this in ilastik)
+
+    example usage:
+
+    ..code block:: python
+        >>> ilastik_path = './ilastik-release/run_ilastik.sh'
+        >>> ilastik_project_path = './data/GCaMP.ilp'
+        >>> simple_segmentation = ilastik_segment(data, ilastik_path, ilastik_project_path)
+        >>>
+    """
+
+    if ilastik_project_path is None:  # defaults to any .ilp file it finds in the home directory
```

Comment:
this is rather error prone, what happens if there are multiple ilp files in the current working directory? I'd remove the default argument and always require an explicit path. 

> python: explicit>implicit